### PR TITLE
Update apigentools, OpenAPI generator and apply patches

### PR DESCRIPTION
**Pull request by umbynos**
The goal of this PR is to update apigentools to 1.0.0 and OpenAPI generator to 4.3.1. This should fix some bugs and problems with the documentation.
The `config` changes format (from json to yaml)
Because of the update, it is required to create new patches

### CODE REVIEW

File: .github/workflows/generate.yaml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
         run: apigentools validate
 
       - name: Render upstream templates
-        run: apigentools templates -p template-patches openapi-git
+        run: apigentools templates
```
Comment (eclipse1985) [inline .github/workflows/generate.yaml:36]: Can you explain a little bit why this change is needed?

File: config/config.yaml [hunk 2]
```diff
@@ -0,0 +1,95 @@
+languages:
```
Comment (eclipse1985) [inline config/config.yaml:1]: Changing from `config.json` to `config.yaml` is just a format change, or there's something different/new in the configuration?

File: .github/workflows/generate.yaml (hunk 1 above)
Comment (rsora) [reply .github/workflows/generate.yaml:36]: [Actually this step can be completely removed ](https://github.com/arduino/clients-iot-api/pull/53/files#diff-7e093faf90eb45ac143b4e20c233deaeL35-L37), as the ver 1.0.0 of `apigentools` handles the templates download in the configuration file as you can see here:
https://github.com/arduino/clients-iot-api/pull/51/files#diff-8bbac5be38be0acd514047133a3900d7R11-R16

Comment (eclipse1985) [reply .github/workflows/generate.yaml:36]: If it's not needed anymore, I'd remove it

File: config/config.yaml (hunk 2 above)
Comment (rsora) [reply config/config.yaml:1]: Updating to version 1.0.0 of `apigentools` determined a "migration" of the json configuration file to a yaml with a new format, [Here the related documentation about the new config format](https://apigentools.readthedocs.io/en/latest/spec_repo/#configconfigyaml).

Additionally this new version does not require an installation of the wrapped Open API tooling as it uses an ad-hoc container that packages the tooling.

File: .github/workflows/generate.yaml (hunk 1 above)
Comment (rsora) [reply .github/workflows/generate.yaml:36]: We did it in https://github.com/arduino/clients-iot-api/pull/53
